ops/pebble.py

```diff
@@ -1466,45 +1467,31 @@ def pull(self, path: str, *, encoding: str = 'utf-8') -> typing.Union[typing.Bin
         if not boundary:
             raise ProtocolError('invalid boundary {!r}'.format(boundary))
 
-        # We have to manually write the Content-Type with boundary, because
-        # email.parser expects the entire multipart message with headers.
-        parser = email.parser.BytesFeedParser()
-        parser.feed(b'Content-Type: multipart/form-data; boundary='
-                    + boundary.encode('utf-8') + b'\r\n\r\n')
+        parser = MultipartFileParser(boundary)
 
-        # Then read the rest of the response and feed it to the parser.
         while True:
```

Comment:
That is a good suggestion.  We do have the name as soon as we read the part header - which precedes the file data - so it should be possible to do something like this.  Perhaps it would be best to create a separate issue for push/pull performance evaluation to facilitate discussion and tracking for these kinds of improvements (see #684).